### PR TITLE
Split Reader/Writer and CloseableReader/CloseableWriter

### PR DESCRIPTION
**Pull request by floitsch**
It didn't make sense that all readers/writers have a `close` method.

### CODE REVIEW

File: lib/io.toit [hunk 1]
```diff
@@ -123,6 +111,19 @@ abstract class Writer:
   // This is a protected method. It should not be "private".
   abstract try-write_ data/Data from/int to/int -> int
 
+
```
Comment (kasperl) [inline lib/io.toit:114]: Too many newlines.

File: lib/tls/socket.toit [hunk 2]
```diff
@@ -13,7 +13,7 @@ import .certificate
 /**
 TLS Socket implementation that can upgrade a TCP socket to a secure TLS socket.
```
Comment (kasperl) [inline lib/tls/socket.toit:14]:
```suggestion
TLS socket implementation that can upgrade a TCP socket to a secure TLS socket.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: lib/tls/socket.toit (hunk 2 above)
Comment (floitsch) [reply lib/tls/socket.toit]: done.

File: lib/io.toit (hunk 1 above)
Comment (floitsch) [reply lib/io.toit]: done.

**Review by floitsch — COMMENTED** [review]
Also removed the strategies (as suggested in the earlier PR).

**Pull request merged**